napari/layers/shapes/shapes.py

```diff
@@ -545,25 +572,27 @@ def __init__(
         )
 
         # set the current_* properties
-        if len(data) > 0:
-            self._current_edge_color = self.edge_color[-1]
-            self._current_face_color = self.face_color[-1]
-        elif len(data) == 0 and len(self.properties) > 0:
-            self._initialize_current_color_for_empty_layer(edge_color, 'edge')
-            self._initialize_current_color_for_empty_layer(face_color, 'face')
-        elif len(data) == 0 and len(self.properties) == 0:
-            self._current_edge_color = transform_color_with_defaults(
-                num_entries=1,
-                colors=edge_color,
-                elem_name="edge_color",
-                default="black",
-            )
-            self._current_face_color = transform_color_with_defaults(
-                num_entries=1,
-                colors=face_color,
-                elem_name="face_color",
-                default="black",
-            )
+        # if len(data) > 0:
```

Comment:
@kevinyamauchi : did you comment this out? If so, do you remember why? I don't remember doing it, but can check! Can also uncomment and see what breaks.

Reply:
I uncommented this and fixed it by using `len(self.features.columns) == 0` instead of `self.features.empty`. Note that `DataFrame.empty` is `True` if there are either 0 rows or 0 columns.

I also had to remove the two lines you added calling `_initialize_current_color` and its definition.

Reply:
Ah sorry, that was me. 😬  Thanks for the fix!